Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -205,142 +205,124 @@ extension NavigationMapView {
-             percentage point will be 1.0, to represent 100%.
-             */
-            if index == congestionSegments.endIndex - 1 {
-                let segmentEndPercentTraveled = CGFloat(1.0)
-                gradientStops[segmentEndPercentTraveled.nextDown] = associatedCongestionColor
-                continue
+            gradientStops[0.0] = traversedRouteColor
+            let currentGradientStop = Double(CGFloat(fractionTraveled).nextDown)
+            if currentGradientStop >= 0.0 {
+                gradientStops[currentGradientStop] = traversedRouteColor
             }
-            
-            /**
-             If this is not the first or last congestion segment, then
-             the starting and ending percent values traveled for this segment
-             will be a fractional amount more/less than the actual values.
-             */
-            let segmentStartPercentTraveled = CGFloat(distanceTraveled / route.distance)
-            gradientStops[segmentStartPercentTraveled.nextUp] = associatedCongestionColor
-            
-            distanceTraveled = distanceTraveled + distance
-            
-            let segmentEndPercentTraveled = CGFloat(distanceTraveled / route.distance)
-            gradientStops[segmentEndPercentTraveled.nextDown] = associatedCongestionColor
-            
-            if index + 1 < congestionSegments.count {
-                gradientStops[segmentEndPercentTraveled.nextUp] = congestionColor(for: congestionSegments[index + 1].properties?["congestion"] as? String, isMain: isMain)
+            gradientStops[fractionTraveled] = minimumSegment.1
+        } else {
+            let percentTraveled = CGFloat(fractionTraveled)
+            gradientStops[0.0] = traversedRouteColor
+            if percentTraveled.nextDown >= 0.0 {
+                gradientStops[Double(percentTraveled.nextDown)] = traversedRouteColor
             }
+            gradientStops[Double(percentTraveled)] = routeCasingColor
         }
         
-        let percentTraveled = CGFloat(fractionTraveled)
-        
-        // Filter out only the stops that are greater than or equal to the percent of the route traveled.
-        var filteredGradientStops = gradientStops.filter { key, value in
-            return key >= percentTraveled
-        }
-        
-        // Then, get the lowest value from the above and fade the range from zero that lowest value,
-        // which represents the % of the route traveled.
-        if let minStop = filteredGradientStops.min(by: { $0.0 < $1.0 }) {
-            filteredGradientStops[0.0] = traversedRouteColor
-            filteredGradientStops[percentTraveled.nextDown] = traversedRouteColor
-            filteredGradientStops[percentTraveled] = minStop.value
-        }
-        
-        var resultGradientStops = [Double: UIColor]()
-
-        filteredGradientStops.filter({ $0.0 >= 0.0 }).forEach {
-            resultGradientStops[Double($0.0)] = $0.1
-        }
```

Comment:
These calculations were removed to slightly improve performance. For example, in new implementation we keep minimum gradient stop is calculated right away, without having additional loop. Whole dictionary copy is not needed as well.